### PR TITLE
feat(core+cli): #347 — init emits annotated crap.example.toml + sync/round-trip/doc-completeness tests

### LINKED ISSUE #347: config: generate crap-rs.example.toml via crap4rs init + sync test
## Summary

Extend `crap4rs init` to emit a fully-annotated config covering every option, commit that output as `crap-rs.example.toml`, and add a sync test asserting `init`-output == committed example — so the exhaustive reference is deterministically generated and can never rot ("documentation rots; CI doesn't").

## Acceptance Criteria

- [ ] `crap4rs init` emits a fully-commented config exercising every supported field/section (`threshold`, `preset`, `metric`, `src`, `exclude`, `[[overrides]]`, `[views.<name>]`, `[output]`, plus per-language sections once (d) lands)
- [ ] The generated output is committed as `crap-rs.example.toml` at repo root
- [ ] A test asserts the committed `crap-rs.example.toml` is byte-identical to fresh `init` output (the rot guard)
- [ ] The example is linked from README/docs as the canonical option reference
- [ ] Distinct from the real minimal `crap-rs.toml` (the dogfood config from (b)) — the example is exhaustive docs, not loaded by the tool

## Context

- Epic: #334
- Blocked-by: #345 (needs the unified name + schema settled)
- Christopher decided (2026-05-29): generate deterministically via `init`, NOT hand-roll (rot risk).
- Code: `crates/crap-core/src/cli/init.rs` (the scaffolder)
- Rule precedent: `feedback_documentation-rots-ci-doesnt`

## Discovery

- How the annotations/comments are carried in `init`'s output (the scaffolder must emit comments, not just values).
- Should the example include the multi-language sections from (d)? (Soft dependency — if (d) lands first, the example covers per-language; otherwise the example grows when (d) lands.)
- Whether the sync test regenerates + diffs, or snapshots.

## Recommended design (2026-05-29 — validate at `/shape`, not locked)

Christopher's intent: **broadly idiomatic** (Rust + software-engineering at large, NOT "matches our repo"), optimized for a **purely agent-built** codebase — so the design should make the wrong thing fail to compile, the rest fail a test loudly, and leave no parallel artifact an agent can forget to update.

**Lean: the type-driven "doc-gen" gold standard** (rustfmt `--print-config`, starship doc-gen) — the config struct is the single source of truth; the example is *derived* from it. **Rejected: askama/templating** (the prose would live in a separate `.jinja` divorced from the field — the exact drift surface we're eliminating; askama is a web/text engine, off-label here). **Rejected: hand-rolled** (rot risk, Christopher's original concern).

Design shape:

- **Single source = the config struct.** Each field's annotation is its own `///` doc comment — one location, which *also* feeds docs.rs + IDE hovers. "Document the field" and "write the example annotation" become the same act in the same place.
- **Surface the doc comments mechanically** via the `documented` crate (derives access to `///` text as associated constants) *or* a tiny local proc-macro. ⚠️ Validate the crate's API + MSRV-1.93 compat before locking (per `feedback_qplan_validate_tooling_before_lock`).
- **Emit via `toml_edit`** (the idiomatic comment-carrying TOML crate — what cargo uses internally): serialized example value + the field's doc attached as a leading TOML comment. Requires adding `#[derive(Serialize)]` to the config types (they derive `Deserialize` only today).
- **Compile-time completeness:** the emitter assembles its field set via an *exhaustive destructure* `let FileConfig { threshold, …, output } = …;` with **no `..`** → adding a config field won't compile until it's wired.
- **One shared fn:** `render_example_config() -> String` in crap-core (adapters layer, next to `config.rs`), called by both `crap4rs init` and `crap4ts init`. `init` writes it verbatim; the committed `crap-rs.example.toml` **is** this fn's output.
- **`init` emits the FULL annotated config** (the `cargo deny init` pattern), not a trimmed starter — this is what makes `init`-output == committed-example *by construction*, which is the only thing that gives the sync test meaning. (The repo's real dogfood `crap-rs.toml` from (b) is a separate, human-trimmed subset.)

Guarantees, attributed precisely (each drift vector → one loud guard):

| Drift | Guard | Layer |
|---|---|---|
| Add a config field, forget to wire it | exhaustive destructure, no `..` | **compile** |
| Field renamed/removed, generator still references it | type error in the destructure/emitter | **compile** |
| Field wired but missing from rendered output | round-trip parse test → field comes back `None` | test |
| Example documents a field that no longer exists | `#[serde(deny_unknown_fields)]` (already on `RawConfig`) in that parse | test |
| Committed example goes stale vs generator | sync test (`init` == committed, by construction) | test |
| Annotation prose drifts from the field | **designed out** — it's the field's own `///`, single location | — |

**Round-trip test correctness note:** assert the parsed example has **every `Option` = `Some` and every collection non-empty** (recursively into `OutputConfig` + each `[views.*]`). Do NOT assert `parsed == FileConfig::default()` — an exhaustive example deliberately shows *non-default* values, so a `== default` check would fail on a correct example and pass on an empty one (backwards).

**Lighter alternative (legitimate fork, not a slam dunk):** `include_str!` a checked-in template, `init` copies it verbatim (cargo-deny style). Broadly idiomatic, least code — but trades the two compile-time guards for runtime-tests-only (no compile-time link to the struct). Given the agent-build constraint, the type-driven path's compile-time coupling is worth the extra setup; decide at `/shape`.

**Cross-link to (d) #348:** the same annotated struct also drives a `schemars` JSON Schema (editor `$schema` autocomplete/validation) — see #348. One source, three derived consumers (docs.rs, annotated example, JSON schema).

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

The second half of the config-type foundation (epic #334). PR1 (#348) landed the annotated `ConfigSchema` + JSON schema + per-language sections + `src` widening; this derives the exhaustive **annotated TOML example** from that same type, so the human-facing config reference can never rot ("documentation rots; CI doesn't"). **Crap-core + CLI only — no `crap4rs/src` change.**

## What's new

- **`render_example_config(&AdapterMeta) -> String`** (crap-core adapters, next to `config.rs`). Hand-builds a `toml_edit::DocumentMut`: **exhaustively destructures every schema struct** (`ConfigSchema`, `OutputSchema`, `OverrideSchema`, `ViewPresetSchema`, `LangSchema`) with **NO `..`** — the compile-time completeness guard, so a new field fails to compile until it's wired into the emitter — and attaches each field's `documented` `///` doc as a leading TOML comment. `threshold` is live; `preset` renders as the **commented mutually-exclusive alternative** (top level AND each `[language.*]` section — Gate E, recursive). Map fields (`views`, `language`) emit **by sorted key** so the output is deterministic.
- **Deps:** `documented` 0.9.2 (`#[derive(DocumentedFields)]`) + `toml_edit` **0.22** (the in-tree major via `toml` 0.8 — NOT the 0.25 `cargo add` default, which adds a duplicate major + raises MSRV). No `Serialize` needed — the renderer hand-builds rather than `toml_edit::ser`, so only `DocumentedFields` is added to the schema types. Both validated on MSRV 1.93 by a kept spike; `cargo-deny` green.
- **`crap.example.toml`** committed at the repo root == `render_example_config` output. NOT loaded by the tool — the canonical option reference.
- **`init` rewrite (minimal-starter → full-annotated-dump):** both `crap4rs init` and `crap4ts init` write `render_example_config` verbatim. `init` is now **non-interactive** — the threshold prompt + `src` auto-detect machinery is removed (there's nothing left to prompt for); the `--non-interactive` CLI flag is retained for back-compat but is **inert**. The existing init tests were rewritten for the full-dump shape.
- **README** links `crap.example.toml` (option reference) + `crap.schema.json` (editor `$schema`); the `init` command row reflects the full-dump.

## Three guards (`tests/config_example_artifact.rs`)

1. **sync** — committed `crap.example.toml` is byte-identical to a fresh render (regenerate-and-diff; the failure message names the fix command).
2. **round-trip** — the rendered example parses with every `Option = Some` **except exactly one of `{preset, threshold}`** (recursive carve-out per `[language.*]`), every collection non-empty, NOT `== default`.
3. **doc-completeness** — every schema field's `///` prose (swept across all five structs via `all_schema_field_docs`) appears in the output, so a field can never render without its annotation.

## Canary handling

Crap-core + CLI only — **no `crap4rs/src` or snapshot file touched** (the `init` wiring goes through the existing `handle_init` signature; the bins' `main.rs` is unchanged). The wire-envelope canary analyzes `../crap4rs/src`, so it is **unaffected — no re-accept**, and stays green in the full run.

## Gates

All green: `cargo fmt --check`; `cargo +1.96.0 clippy --workspace --all-targets -- -D warnings` (CI toolchain 1.96, not local 1.95); `cargo nextest run --workspace --all-targets` (**1301 passed**); `cargo +1.93 check --workspace --all-targets` (MSRV); `cargo deny check`; `bdd-tracked-lint`; `mutants-skip-lint`; AST-purity L4 clean. PR1's `crap.schema.json` sync test passed in the same run — confirming the `DocumentedFields` derive didn't perturb schemars output.

The 14 `@unwired` scenarios in `config_schema.feature` (tracked #169) pin the full #347+#348 contract; this PR's behaviors are verified by the three crap-core guards above + the rewritten `init` tests.

Closes #347

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/crap-core/src/adapters/config.rs [hunk 1]
```diff
@@ -587,6 +590,486 @@ pub fn config_json_schema() -> String {
...
+        T::FIELD_NAMES
+            .iter()
+            .zip(T::FIELD_DOCS.iter())
+            .map(|(name, doc)| {
+                let label: &'static str =
+                    Box::leak(format!("{struct_name}.{name}").into_boxed_str());
+                (label, *doc)
+            })
+            .collect()
+    }
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap-core/src/adapters/config.rs]: Fixed in e44daee. `all_schema_field_docs` now returns `Vec<(String, &'static str)>` — owned labels, no `Box::leak`. The sole caller (the `every_schema_field_doc_appears_in_generated_example` doc-completeness test) only uses the label in `{label}` diagnostics, so `String` is a drop-in. Verified green under `cargo +1.96.0 clippy -p crap-core --all-targets -- -D warnings` and the 3-test example-artifact suite.

File: crates/crap-core/src/adapters/config.rs [hunk 2]
```diff
@@ -587,6 +590,486 @@ pub fn config_json_schema() -> String {
...
+        ])),
+        exclude: Some(
+            meta.default_excludes
+                .iter()
+                .map(|s| (*s).to_string())
+                .collect::<Vec<_>>()
+                .into_iter()
+                .chain(std::iter::once("generated/**".to_string()))
+                .collect(),
+        ),
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap-core/src/adapters/config.rs:1029]: Fixed in e44daee. Dropped the intermediate `collect::<Vec<_>>().into_iter()`; the mapped iterator now chains `std::iter::once("generated/**".to_string())` directly before the final `collect()`. Output is byte-identical — `committed_example_matches_generated` (the sync guard) stays green.

**Comment by cmbays** [comment]
## Review disposition — `e44daee`

Addressed the CI failure plus both bot findings. Each verified against primary source before changing anything.

| Source | Finding | Disposition |
|---|---|---|
| **CI** — `cargo doc (-D warnings)` | `render_example_config` rustdoc linked the **private** `validate_raw_config` via an intra-doc link → `rustdoc::private_intra_doc_links` failed the build | **Fixed.** Rephrased to state the mutual-exclusion behavior directly. A docs.rs reader can't resolve a private symbol anyway, so this also removes an opaque internal reference from public docs. |
| **Gemini** (medium) `config.rs:1063` | `all_schema_field_docs` uses `Box::leak` to mint `'static` labels on every call of a `pub fn` | **Fixed.** Returns owned `Vec<(String, &'static str)>`. The sole caller (doc-completeness test) uses the label only in `{label}` diagnostics. |
| **Gemini** (medium) `config.rs:1031` | `exclude` builder collects an intermediate `Vec` only to `into_iter()` and chain one element | **Fixed.** Chains `std::iter::once(...)` directly onto the mapped iterator. |

**No behavior change.** `committed_example_matches_generated` stays green → `crap.example.toml` is byte-identical (no regeneration). Round-trip + doc-completeness green. No `crap4rs/src` touched → wire-envelope canary unaffected.

Local gates re-run green: `cargo fmt --check`, `cargo +1.96.0 clippy -p crap-core --all-targets -- -D warnings`, `RUSTDOCFLAGS=-D warnings cargo doc --no-deps -p crap-core`, and the `config_example_artifact` suite (3/3).

> CodeRabbit was rate-limited on open (per-dev hourly cap); Gemini provided the review. Will re-request `@coderabbitai review` only if its window reopens before merge.
